mapper/tedge_mapper/src/az_mapper.rs

```diff
@@ -0,0 +1,198 @@
+use crate::error::*;
+use crate::mapper::*;
+use crate::time_provider::TimeProvider;
+use mqtt_client::Topic;
+use thin_edge_json::serialize::ThinEdgeJsonSerializer;
+
+#[derive(Debug)]
+pub struct AzureMapperConfig {
+    pub in_topic: Topic,
+    pub out_topic: Topic,
+    pub errors_topic: Topic,
+}
+
+impl Default for AzureMapperConfig {
+    fn default() -> Self {
+        Self {
+            in_topic: Topic::new("tedge/measurements").unwrap(),
+            out_topic: Topic::new("az/messages/events/").unwrap(),
+            errors_topic: Topic::new("tedge/errors").unwrap(),
+        }
+    }
+}
+
+impl Into<MapperConfig> for AzureMapperConfig {
+    fn into(self) -> MapperConfig {
+        MapperConfig {
+            in_topic: self.in_topic,
+            out_topic: self.out_topic,
+            errors_topic: self.errors_topic,
+        }
+    }
+}
+
+pub struct AzureConverter {
+    pub(crate) add_timestamp: bool,
+    pub(crate) time_provider: Box<dyn TimeProvider>,
+}
+
+impl Converter for AzureConverter {
+    type Error = ConversionError;
+    fn convert(&self, input: &[u8]) -> Result<Vec<u8>, Self::Error> {
+        // Size check
+        let threshold = 255 * 1000; // 255KB
+        let () = is_smaller_than_size_threshold(input, threshold)?;
+
+        // Add timestamp if necessary
+        let default_timestamp = if self.add_timestamp {
+            Some(self.time_provider.now())
+        } else {
+            None
+        };
+
+        let mut serializer = ThinEdgeJsonSerializer::new_with_timestamp(default_timestamp);
+
+        let () = thin_edge_json::json::parse_utf8(input, &mut serializer)?;
+        Ok(serializer.bytes()?)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::az_mapper::AzureConverter;
+    use crate::mapper::Converter;
+    use crate::time_provider::TestTimeProvider;
+    use assert_json_diff::*;
+    use chrono::{FixedOffset, TimeZone};
+    use serde_json::json;
+
+    #[test]
+    fn test_azure_converter_invalid_input() {
```

Comment:
@rina23q Just a side note: All tests test `AzureConverter`. If you'd move `AzureConverter` into it's own module/file, you would not need to prefix the test functions with `azure_converter`. Also if you leave out the `test_` prefix (they are "obvious"), you end up with shorter, more readable test function names.

You could then rename them like:

```rust
struct AzureConverter {...}

#[test]
fn converting_invalid_json_is_invalid() {
   ...
}

#[test]
fn converting_input_with_no_timestamp_produces_output_with_no_timestamp() {
}

#[test]
fn converting_input_with_timestamp_produces_output_with_timestamp_given_add_timestamp_is_false() {
   // ...
}

...
```

Think about tests as facts. You want to prove a fact, that something "holds". Describe the facts, the expected behavior.